stub/src/main/java/io/grpc/stub/ServerCalls.java

```diff
@@ -331,10 +344,17 @@ public void setCompression(String compression) {
     @Override
     public void onNext(RespT response) {
       if (cancelled) {
-        if (onCancelHandler == null) {
-          throw Status.CANCELLED.withDescription("call already cancelled").asRuntimeException();
+        if (serverStreamingOrBidi) {
+          throw Status.CANCELLED
+              .withDescription("call already cancelled. "
+                  + "ServerCallStreamObserver.setOnCancelHandler() disables this exception")
```

Comment:
This message is confusing. It seems a hint to the developer to change their source code: "To disable this exception: use ServerCallStreamObserver.setOnCancelHandler()", not an error message to users.

Reply:
Do you have a suggestion for alternative wording? This is what we discussed in the API review. When would users see this exception?

Reply:
> Do you have a suggestion for alternative wording?

No specific alternatives. I think we should just tell the user that this message is a hint to the developers.

Reply:
Decided on:
"Use ServerCallStreamObserver.setOnCancelHandler() to disable this exception"